compiler/pipes/gen-tree-postprocess.cpp

```diff
@@ -11,6 +11,35 @@
 #include "compiler/gentree.h"
 
 namespace {
+template <typename F>
+bool contains_vertex(const VertexPtr &root, F fn) {
+  if (fn(root)) {
+    return true;
+  }
+  for (const auto &v : *root) {
+    if (contains_vertex(v, fn)) {
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
JFYI: contains_vertex could be implemented this way
```
template <Operation op>
bool contains_vertex(VertexPtr root) {
  return root->type() == op || std::any_of(root->begin(), root->end(), contains_vertex<op>);
}

```

Reply:
I used the proposed code, thanks.